### ISSUE TITLE
Pretty printing `SetLogic(ALL)` gives a match error

### ISSUE THREAD

**Issue by ahmadsalim** [regb/scala-smtlib#16]
I am getting the following error:

```{scala}
scala> semantics.HeapConsistencyChecker.test()
scala.MatchError: ALL (of class smtlib.parser.Commands$ALL$)
  at smtlib.printer.TerminalTreesPrinter$class.printLogic(TerminalTreesPrinter.scala:62)
  at smtlib.printer.RecursivePrinter$.printLogic(RecursivePrinter.scala:10)
  at smtlib.printer.RecursivePrinter$.printCommand(RecursivePrinter.scala:162)
  at smtlib.printer.RecursivePrinter$$anonfun$printScript$1.apply(RecursivePrinter.scala:16)
  at smtlib.printer.RecursivePrinter$$anonfun$printScript$1.apply(RecursivePrinter.scala:15)
  at scala.collection.immutable.List.foreach(List.scala:381)
  at smtlib.printer.RecursivePrinter$.printScript(RecursivePrinter.scala:15)
  at smtlib.printer.Printer$class.toString(Printer.scala:27)
  at smtlib.printer.RecursivePrinter$.toString(RecursivePrinter.scala:10)
  at semantics.HeapConsistencyChecker$.test(HeapConsistencyChecker.scala:12)
  ... 43 elided
```

when trying to run:

```{scala}
object Test {
  import smtlib.parser.Commands._
  import smtlib.parser.Terms._
  import smtlib.theories.Ints._

  def test() {
    val x = QualifiedIdentifier(SimpleIdentifier(SSymbol("x")))
    val y = QualifiedIdentifier(SimpleIdentifier(SSymbol("y")))
    val pf = smtlib.printer.RecursivePrinter.toString(Script(
      List(SetLogic(ALL),
           DeclareFun(SSymbol("x"), Seq(), IntSort()),
           DeclareFun(SSymbol("y"), Seq(), IntSort()),
           Assert(LessThan(NumeralLit(0), Add(x, y))),
           CheckSat()
           ))) //(assert (< 0 (+ x y)))
    print(pf)
  }
}
```

**Comment by regb** [comment]
There was indeed missing implementation for the logic "ALL", this is now fixed in master with commit 401b9efad6873039f66f700b27b8ddb4dafa0280
Thanks for pointing that out

**Comment by ahmadsalim** [comment]
Awesome, thanks for fixing this!
